packages/quantic/force-app/main/default/lwc/quanticResultQuickview/__tests__/quanticResultQuickview.test.js

```diff
@@ -131,41 +135,158 @@ describe('c-quantic-result-quick-view', () => {
+
+      expect(quickViewButton.disabled).toBeTruthy();
+    });
+  });
+
+  describe('when the component has custom properties', () => {
+    it('should render the custom icon correctly', async () => {
+      const customOptions = {
+        result: exampleResult,
+        previewButtonIcon: 'utility:file',
+      };
+      const element = createTestComponent(customOptions);
       await flushPromises();
 
-      expect(functionMocks.pushRecentResult).toHaveBeenCalled();
+      const quickViewIcon = element.shadowRoot.querySelector(selectors.icon);
+
+      expect(quickViewIcon.iconName).toBe('utility:file');
+    });
+
+    it('should render the custom label correctly', async () => {
+      const customOptions = {
+        result: exampleResult,
+        previewButtonLabel: 'Custom Label',
+      };
+      const element = createTestComponent(customOptions);
+      await flushPromises();
+
+      const quickViewButton = element.shadowRoot.querySelector(
+        selectors.quickviewButton
+      );
+
+      expect(quickViewButton.textContent).toContain('Custom Label');
+      expect(quickViewButton.querySelector(selectors.icon).classList).toContain(
+        'slds-button__icon_right'
+      );
+    });
+
+    it.each([['brand'], ['outline_brand'], ['result-action'], ['turbo']])(
+      'should render the custom variant %s',
+      async (testVariant) => {
+        const customOptions = {
+          result: exampleResult,
+          previewButtonVariant: testVariant,
+        };
+        const element = createTestComponent(customOptions);
+        await flushPromises();
+
+        const quickViewButton = element.shadowRoot.querySelector(
+          selectors.quickviewButton
+        );
+
+        const expectedClass =
+          testVariant === 'result-action'
+            ? 'slds-button_icon-border-filled'
+            : `slds-button_${testVariant}`;
+        expect(quickViewButton.classList).toContain(expectedClass);
+      }
+    );
+
+    it('should render the tooltip correctly', async () => {
+      const customOptions = {
+        result: exampleResult,
+        tooltip: 'Custom Tooltip',
+      };
+      const element = createTestComponent(customOptions);
+      await flushPromises();
+
+      const tooltip = element.shadowRoot.querySelector(selectors.tooltip);
+
+      expect(tooltip.textContent).toBe('Custom Tooltip');
+    });
+
+    it('should render the maximum preview size correctly', async () => {
+      const customOptions = {
+        result: exampleResult,
+        maximumPreviewSize: 500,
+      };
+      const element = createTestComponent(customOptions);
+      await flushPromises();
+
+      expect(element.maximumPreviewSize).toBe(500);
     });
   });
 
-  describe('when the component is not used in the search use case', () => {
-    beforeAll(() => {
-      useCase = 'insight';
+  describe('when the quickview is opened and closed', () => {
+    it('should open the quickview modal', async () => {
```

Comment:
When of the things that could be useful to test is to confirm that the `engineId` property used when creating the QuanticResultQuickview component has been successfully passed to the QuanticResultLink component displayed when opening the Quickview.